src/OrchardCore/OrchardCore.ContentManagement.GraphQL/Mutations/Types/CreateContentItemInputType.cs

```diff
@@ -0,0 +1,55 @@
+using System.Collections.Generic;
+using GraphQL.Types;
+using Microsoft.AspNetCore.Http;
+
+namespace OrchardCore.ContentManagement.GraphQL.Mutations.Types
+{
+    public class CreateContentItemInputType : InputObjectGraphType<ContentItem>
+    {
+        public CreateContentItemInputType()
+        {
+            Field(ci => ci.ContentType, false);
+            Field(ci => ci.Author, true);
+            Field(ci => ci.Owner, true);
+
+            Field(ci => ci.Published, true);
+            Field(ci => ci.Latest, true);
+
+            Field<ContentPartsInputType>("contentParts");
+        }
+    }
+
+    public class ContentPartsInputType : InputObjectGraphType
+    {
+        private readonly IHttpContextAccessor _httpContextAccessor;
+
+        public ContentPartsInputType(
+            IHttpContextAccessor httpContextAccessor,
+            IEnumerable<ContentPart> contentParts,
+            ITypeActivatorFactory<ContentPart> typeActivator)
+        {
+            Name = "ContentPartsInput";
+
+            foreach (var part in contentParts)
+            {
+                var partName = part.GetType().Name;
+                var activator = typeActivator.GetTypeActivator(partName);
+
+                var inputGraphType =
+                    typeof(InputObjectGraphType<>).MakeGenericType(activator.Type);
+
+                var inputGraphTypeResolved = (IInputObjectGraphType)httpContextAccessor.HttpContext
```

Comment:
maybe better to use the field `_httpContextAccessor` if you initialize it before (not at the end), or just not declare it.